plugins/redfish/fu-plugin-redfish.c

```diff
@@ -181,6 +185,89 @@ fu_redfish_plugin_discover_smbios_table(FuPlugin *plugin, GError **error)
 	return TRUE;
 }
 
+#ifdef HAVE_LINUX_IPMI_H
+static gchar *
+fu_common_generate_password(guint length)
+{
+	GString *str = g_string_sized_new(length);
+
+	/* get a random password string */
+	while (str->len < length) {
+		gchar tmp = (gchar)g_random_int_range(0x0, 0xff);
+		if (g_ascii_isalnum(tmp))
+			g_string_append_c(str, tmp);
+	}
+	return g_string_free(str, FALSE);
+}
+
+static gboolean
+fu_redfish_plugin_ipmi_create_user(FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data(plugin);
+	const gchar *username_fwupd = "fwupd";
+	guint8 user_id = 0x04;
```

Comment:
looks like you hardcoded the user_id = 0x04 ..... you probably need to add some logic to handle cases where 0x04 is used by other users, where you need to pick another channel number for user_id.

Reply:
Yup, certainly a valid criticism. I guess the best thing to do at startup is:

 * look from `user_id` 2 to 63 for a user called fwupd
 * if found, report an error as we didn't save the generated password and can't delete the user
 * if not found, create a fwupd user in the first unused slot and continue to enumerate

@tcap26 does that sound [more] sane?